[AI-5281] Barracuda Secure Edge Integration PR (#20423)
DataDog/integrations-core: barracuda_secure_edge/README.md

    - box_Firewall, box_Firewall_Activity -> log_component:firewall
    - box_Network_dhcp -> log_component:dhcp
    - box_Network_activation -> log_component:network
    - box_System_mgmaccess -> log_component:access
    - box_SSH_sshd, box_SSH_config -> log_component:ssh
    - srv_CSC_VPN, srv_CSC_VPN_sslvpn -> log_component:vpn   
    ```

3. [Restart the Agent][3].

### Installation

The barracuda_secure_edge check is included in the [Datadog Agent][2] package.

### Prerequisites

1. Administrative access to Barracuda Secure Edge installed on your server.
2. The Datadog Agent installed and running (on a server or container that can receive syslog messages).
3. Network Access between the firewall and the Datadog Agent (usually port 514, but may be a custom value).
4. Syslog support enabled in the Datadog Agent (with a TCP or UDP  listener configured).

### Validation

1. Confirm the Datadog Agent is listening on the correct port (`514` in the following examples):

    `sudo netstat -tunlp | grep 514`

    If using TCP and UDP listeners, use the following command:

    `sudo lsof -i :514`

2. Confirm logs are reaching the Agent from the correct log source:

>     `tail -f /var/phion/logs/*.log`

Reviewer: Note that the example paths should be consistent throughout the doc. This is different from the paths used in the Setup section.

    **Note**: If the file doesn't exist, verify that syslog logs are being written by your configuration.

3. Use the tcpdump command to confirm network traffic on the Datadog Agent host:

    `sudo tcpdump -i any port 514`

After running this command, you should see traffic from the Secure Edge IP address. If you don't see any such traffic, check the firewall rules between Secure Edge and the Datadog Agent. Confirm the correct protocol (UDP or TCP) is being used on both sides.
4. Check the Datadog [Live Tail][5] for logs from the source and service you defined in the `conf.yaml` file.
5. After following these steps, you can create a test log on the firewall by triggering an event.
6. Check for tags or facets to use for better filtering based on the required data.

## Data Collected

### Metrics

The Barracuda_Secure_Edge integration does not include any metrics.

> ### Logs

Reviewer: While the pieces of information are there, I think the connection between the log source requirements and actual log files is missing.

Currently, we show required tags and an example of tailing a file, but we don’t explain how logs are routed into those files.

We should add an explicit prerequisite: configure a syslog service to split Barracuda Secure Edge logs into separate files. This ensures the paths in the Agent configuration actually exist and receive the correct logs.

# Agent Integration: barracuda_secure_edge

## Overview

> This integration monitors [barracuda_secure_edge][4].

Reviewer: ```suggestion
Barracuda Secure Edge is a unified Secure Access Service Edge (SASE) platform that includes Next-Generation Firewall (NGFW), zero trust, and secure Software-Defined Wide Area Network (SD-WAN) capabilities. This integration allows you to collect and analyze logs from your [barracuda_secure_edge][4] deployment to monitor security events, network traffic, and system activity.
```

## Setup
1. Collecting logs is disabled by default in the Datadog Agent. Enable it in the `datadog.yaml` file with:

    ```yaml
      logs_enabled: true
    ```
2. Add this configuration block to your `secure_edge.d/conf.yaml` file to start collecting your Secure Edge logs:

    ```yaml
      logs:

>         - type: file

Reviewer: It seems like we are mixing file-based and syslog methods. The setup steps use a file, but the validation steps use syslog? My understanding is we should be consistent here.

          path:  /var/log/secure_edge.log
          source: secure_edge
          service: <SERVICE_NAME>
    ```

    Change the `path` and `service` parameter values for your environment.

3. [Restart the Agent][3].

### Installation

The barracuda_secure_edge check is included in the [Datadog Agent][2] package.

### Prerequisites

1. Administrative access to Barracuda Secure Edge installed on your server.
2. The Datadog Agent installed and running (on a server or container that can receive syslog messages).
3. Network Access between the firewall and the Datadog Agent (usually port 514, but may be a custom value).

> 4. Syslog support enabled in the Datadog Agent (with a TCP or UDP  listener configured).

Reviewer: This should be the first subsection of Setup, so users see this info before starting.

### Validation

1. Confirm the Datadog Agent is listening on the correct port (`514` in the following examples)
    `sudo netstat -tunlp | grep 514`
    If using TCP and UDP listeners, use the following command:
    `sudo lsof -i :514`
2. Confirm logs are reaching the Agent from the correct log source.
    `tail -f /var/phion/logs/*.log`
**Note**: If the file doesn't exist, verify that syslog logs are being written by your configuration.
3. Use the tcpdump command to confirm network traffic. On the Datadog Agent host:
    `sudo tcpdump -i any port 514`
After running this command, you should see traffic from the Secure Edge IP address. If you don't see any such traffic, check the firewall rules between Secure Edge and the Datadog Agent. Confirm the correct protocol (UDP or TCP) is being used on both sides.
4. Check the Datadog [Live Tail][5] in Datadog for logs from the source and service you defined in the `conf.yaml` file.
5. After following these steps, you can create a test log on the firewall by triggering an event.
6. Check for tags or facets to use them for better filtering based on the required data.

> ### Metrics
> 
> Barracuda_Secure_Edge does not include any metrics.
> 
> 
> ### Log Collection
> ## Data Collected
> The Barracuda Secure Edge logs contain key information such as the event timestamp, source and destination IPs and ports, protocol used, firewall action (allow/deny), the matched rule name, user identity (if available), log type (e.g., firewall, VPN, authentication), network interface, device name, and status of the operation, all of which help monitor traffic behavior, access control, and system activity and many more which are collected by DataDog.
> 
> ### Events
> 
> The Secure_edge integration includes log events such as failed logins and rule hits.

Reviewer: ```suggestion
## Data Collected

### Metrics

Barracuda_Secure_Edge does not include any metrics.

### Events

The Barracuda Secure Edge integration does not include any events.

### Logs

The Barracuda Secure Edge integration collects logs containing the following types of information:

- **Security Events**: Firewall actions (allow/deny), rule matches, and security policy violations
- **Network Traffic**: Source and destination IPs/ports, protocols, and network interfaces
- **Authentication**: User login attempts, successes, and failures
- **VPN Activity**: VPN connection events and status
- **System Events**: Device status, configuration changes, and system health

```

## Troubleshooting

Need help? Contact [Datadog support][1].

[1]: https://docs.datadoghq.com/help/

> [2]: https://app.datadoghq.com/account/settings/agent/latest

Reviewer: ```suggestion
[2]: /account/settings/agent/latest
```

[3]: https://docs.datadoghq.com/agent/guide/agent-commands/#start-stop-and-restart-the-agent
[4]: https://www.barracuda.com/products/network-protection/secureedge

> [5]: https://app.datadoghq.com/logs/livetail

Reviewer: ```suggestion
[5]: /logs/livetail
```

# Agent Integration: barracuda_secure_edge

## Overview

Barracuda Secure Edge is a unified Secure Access Service Edge (SASE) platform that includes Next-Generation Firewall (NGFW), zero trust, and secure Software-Defined Wide Area Network (SD-WAN) capabilities. This integration allows you to collect and analyze logs from your [barracuda_secure_edge][4] deployment to monitor security events, network traffic, and system activity.

## Setup
### Prerequisites

1. Administrative access to Barracuda Secure Edge installed on your server.
2. The Datadog Agent installed and running (on a server or container that can receive syslog messages).

> 3. Network Access between the firewall and the Datadog Agent (usually port 514, but may be a custom value).

Reviewer: ```suggestion
3. Network access between the firewall and the Datadog Agent (usually port 514, but may be a custom value).
```

> 1. Administrative access to Barracuda Secure Edge installed on your server.
> 2. The Datadog Agent installed and running (on a server or container that can receive syslog messages).
> 3. Network Access between the firewall and the Datadog Agent (usually port 514, but may be a custom value).
> 4. Syslog support enabled in the Datadog Agent (with a TCP or UDP  listener configured).

Reviewer: ```suggestion
- Administrative access to Barracuda Secure Edge installed on your server.
- The Datadog Agent installed and running (on a server or container that can receive syslog messages).
- Network Access between the firewall and the Datadog Agent (usually port 514, but may be a custom value).
- Syslog support enabled in the Datadog Agent (with a TCP or UDP  listener configured).
```

> ### Setup Configurations

Reviewer: ```suggestion
### Setup configuration
```
